test/src/test/java/org/apache/accumulo/test/InMemoryMapIT.java

```diff
@@ -0,0 +1,320 @@
+    }
+    Mutation m3 = new Mutation("b");
+    for (int i = 1; i < 3; i++) {
+      m3.put(new Text("8cf" + i), new Text("8cq" + i), new Value(Integer.toString(i).getBytes()));
+    }
+
+    assertEquivalentMutate(Arrays.asList(m1, m2, m3));
+  }
+
+  private void assertEquivalentMutate(Mutation m) {
+    assertEquivalentMutate(Collections.singletonList(m));
+  }
+
+  private void assertEquivalentMutate(List<Mutation> mutations) {
+    InMemoryMap defaultMap = null;
+    InMemoryMap nativeMapWrapper = null;
+    InMemoryMap localityGroupMap = null;
+    InMemoryMap localityGroupMapWithNative = null;
+
+    try {
+      defaultMap = new InMemoryMap(false, tempFolder.newFolder().getAbsolutePath());
+      nativeMapWrapper = new InMemoryMap(true, tempFolder.newFolder().getAbsolutePath());
+      localityGroupMap = new InMemoryMap(getLocalityGroups(), false, tempFolder.newFolder().getAbsolutePath());
+      localityGroupMapWithNative = new InMemoryMap(getLocalityGroups(), false, tempFolder.newFolder().getAbsolutePath());
+    } catch (IOException e) {
+      log.error("Error getting new InMemoryMap ", e);
+      fail(e.getMessage());
+    }
+
+    defaultMap.mutate(mutations);
+    nativeMapWrapper.mutate(mutations);
+    localityGroupMap.mutate(mutations);
+    localityGroupMapWithNative.mutate(mutations);
+
+    // let's use the transitive property to assert all four are equivalent
+    assertMutatesEquivalent(mutations, defaultMap, nativeMapWrapper);
+    assertMutatesEquivalent(mutations, defaultMap, localityGroupMap);
+    assertMutatesEquivalent(mutations, defaultMap, localityGroupMapWithNative);
+  }
+
+  /**
+   * Assert that a set of mutations mutate to equivalent map in both of the InMemoryMaps.
+   * <p>
+   * In this case, equivalent means 2 things.
+   * <ul>
+   * <li>The size of both maps generated is equal to the number of key value pairs in all mutations passed</li>
+   * <li>The size of the map generated from the first InMemoryMap equals the size of the map generated from the second</li>
+   * <li>Each key value pair in each mutated map has a unique id (kvCount)</li>
+   * </ul>
+   *
+   * @param mutations
+   *          List of mutations
+   * @param imm1
+   *          InMemoryMap to compare
+   * @param imm2
+   *          InMemoryMap to compare
+   */
+  private void assertMutatesEquivalent(List<Mutation> mutations, InMemoryMap imm1, InMemoryMap imm2) {
+    int mutationKVPairs = countKVPairs(mutations);
+
+    List<MemKey> memKeys1 = getArrayOfMemKeys(imm1);
+    List<MemKey> memKeys2 = getArrayOfMemKeys(imm2);
+
+    assertEquals("Not all key value pairs included: " + dumpInMemoryMap(imm1, memKeys1), mutationKVPairs, memKeys1.size());
+    assertEquals("Not all key value pairs included: " + dumpInMemoryMap(imm2, memKeys2), mutationKVPairs, memKeys2.size());
+    assertEquals("InMemoryMaps differ in size: " + dumpInMemoryMap(imm1, memKeys1) + "\n" + dumpInMemoryMap(imm2, memKeys2), memKeys1.size(), memKeys2.size());
```

Comment:
I think it will always be true.  I initially did it this way to give a very specific message if something was wrong.  But you are correct, if the InMemoryMaps differ in size, one of the two prior asserts will fail first.